engines/grim/gfx_opengl.cpp

```diff
@@ -1683,29 +1703,11 @@ void GfxOpenGL::prepareMovieFrame(Graphics::Surface *frame) {
 	GLenum dataType;
 	int bytesPerPixel = frame->format.bytesPerPixel;
 
-	// Aspyr Logo format
-	if (frame->format == Graphics::PixelFormat(4, 8, 8, 8, 0, 8, 16, 24, 0)) {
-#if !defined(__amigaos4__)
-		format = GL_BGRA;
-		dataType = GL_UNSIGNED_INT_8_8_8_8;
-#else
-		// MiniGL on AmigaOS4 doesn't understand GL_UNSIGNED_INT_8_8_8_8 yet.
-		format = GL_BGRA;
+	// Used by Bink, QuickTime, MPEG, Theora and paletted SMUSH
+	if (frame->format == getMovieFormat()) {
+		format = GL_RGBA;
 		dataType = GL_UNSIGNED_BYTE;
-#endif
-	// Used by Grim Fandango Remastered
-	} else if (frame->format == Graphics::PixelFormat(4, 8, 8, 8, 8, 8, 16, 24, 0)) {
-#if !defined(__amigaos4__)
-		format = GL_BGRA;
-		dataType = GL_UNSIGNED_INT_8_8_8_8;
-#else
-		// MiniGL on AmigaOS4 doesn't understand GL_UNSIGNED_INT_8_8_8_8 yet.
-		format = GL_BGRA;
-		dataType = GL_UNSIGNED_BYTE;
-#endif
-	} else if (frame->format == Graphics::PixelFormat(4, 8, 8, 8, 0, 16, 8, 0, 0)) {
-		format = GL_BGRA;
-		dataType = GL_UNSIGNED_INT_8_8_8_8_REV;
```

Comment:
@ccawley2011 can you explain this change?

We had the following pixel formats:
- Graphics::PixelFormat(4, 8, 8, 8, 0, 8, 16, 24, 0) - Used by EMI full
- Graphics::PixelFormat(4, 8, 8, 8, 8, 8, 16, 24, 0) - Not sure?
- Graphics::PixelFormat(4, 8, 8, 8, 8, 8, 16, 24, 0) - The comment mentions Grim Remastered, but that's not true - it's using the format you added, i.e. Graphics::PixelFormat(4, 8, 8, 8, 8, 0, 8, 16, 24)
- Graphics::PixelFormat(4, 8, 8, 8, 8, 16, 8, 0, 24) - Not sure?
- Graphics::PixelFormat(4, 8, 8, 8, 0, 16, 8, 0, 0) - Not sure?

and now we have ended with the following formats:
- Graphics::PixelFormat(4, 8, 8, 8, 8, 0, 8, 16, 24) - used by Grim Remastered, EMI demo
- Graphics::PixelFormat(4, 8, 8, 8, 0, 8, 16, 24, 0) - missing, used by EMI Full

There's also this one for SMUSH, which I'm mentioning for completeness. It's used by the original version of Grim:
- Graphics::PixelFormat(2, 5, 6, 5, 0, 11, 5, 0, 0)

So, the question is... why have these formats been removed, and should we also add the EMI Full format too?

Reply:
To elaborate, codecs that normally output YUV graphics can have the output format specified to avoid extra pixel format conversion. Previously, the Bink, MPEG, QuickTime and Theora players had somewhat arbitrary formats specified, but this PR changes it so that all of them use a single pixel format that is reported as ideal by the renderer.

Reply:
@ccawley2011 Thanks for clearing out my questions!